internal/pkg/api/handleCheckin.go

```diff
@@ -384,6 +386,8 @@ func (ct *CheckinT) resolveSeqNo(ctx context.Context, zlog zerolog.Logger, req C
 			if errors.Is(err, dl.ErrNotFound) {
 				zlog.Debug().Str("token", *ackToken).Msg("revision token not found")
 				err = nil
+				// should be left the ActionSeqNo if no ackToken, otherwise would be overwritten with 0 on a Fleet Server restart
+				return seqno, nil
```

Comment:
without the return, the code returned `seqno = []int64{sn}`, and `sn` had the value 0 if `ackToken` not defined.